### PR TITLE
feat: add --template flag for addon generator

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**
Enhancement

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Yes

**Summary**
Add `--template` flag for `loader` and `plugin` commands.

**Does this PR introduce a breaking change?**
Nope

**Other information**
N/A

### CODE REVIEW

**Review by alexander-akait — APPROVED** [review]
Let's fix CI

**Review by alexander-akait — APPROVED** [review]
/cc @webpack/cli-team need review

File: packages/generators/src/addon-generator.ts [hunk 1]
```diff
@@ -33,21 +36,44 @@ const addonGenerator = (
...
         }
 
         public props: Generator.Question;
         public copy: (value: string, index: number, array: string[]) => void;
         public copyTpl: (value: string, index: number, array: string[]) => void;
 
-        public prompting(): Promise<void> {
+        public async prompting(): Promise<void> {
+            if (!this.supportedTemplates.includes(this.template)) {
+                this.utils.logger.log(`${yellow(`⚠ ${this.template} is not a valid template, please select one from below`)}`);
```
[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/generators/src/addon-generator.ts:62]: It is the case currently:
https://github.com/webpack/webpack-cli/blob/feat/addon-generator/packages/generators/src/addon-generator.ts#L51

[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/generators/src/addon-generator.ts:62]: It is the case currently:
https://github.com/webpack/webpack-cli/blob/feat/addon-generator/packages/generators/src/addon-generator.ts#L51

[the remark this answers is not in the document]
Comment (anshumanv) [reply packages/generators/src/addon-generator.ts:62]: I mean https://github.com/webpack/webpack-cli/blob/f7251217c541026e1c7f0b26d0cb34d7019903f5/packages/webpack-cli/lib/utils/logger.js#L6

File: packages/generators/__tests__/plugin-generator.test.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ describe('plugin generator', () => {
             .withPrompts({
                 name: pluginName,
             })
-            .withOptions({ cli: { utils } });
+            .withOptions({ cli: { utils }, options: { template: 'default' } });
```
Comment (alexander-akait) [inline packages/generators/__tests__/plugin-generator.test.ts:14]: Can we have two tests with `template` and without?

**Review by alexander-akait — APPROVED** [review]
Thanks

**Review by anshumanv — APPROVED** [review]

**Pull request merged**